### PR TITLE
Code cleanup and binary removal

### PR DESCRIPTION
**Pull request by undef21**
- remove -Wextra-semi build option as unsupported in gcc 6
- remove binary src/dist/fheroes2 (added with PR #3741)
- remove unused System::SetEnvironment
- fix building src/fheroes2/agg/bin_info.h with gcc6 (missed size_t definition)

### CODE REVIEW

File: src/fheroes2/agg/bin_info.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #ifndef H2BIN_FRM_H
 #define H2BIN_FRM_H
 
+#include <stddef.h>
```
Comment (qarkai) [inline src/fheroes2/agg/bin_info.h:24]:
```suggestion
#include <cstdint>
```

Comment (ihhub) [reply src/fheroes2/agg/bin_info.h:24]: Yes, please. We need to use C++ headers over C.

Comment (undef21) [reply src/fheroes2/agg/bin_info.h]: Oh, sorry. Error is "error: ‘size_t’ does not name a type", not uint32_t. Now used <cstddef>.

File: src/Makefile [hunk 2]
```diff
@@ -35,7 +35,7 @@ CFLAGS_TP := $(CFLAGS)
 CXXFLAGS_TP := $(CXXFLAGS)
 
 # Add -Wconversion -Wsign-conversion -Weffc++ -Woverloaded-virtual flags back once we fix all other warnings!
```
Comment (ihhub) [inline src/Makefile:37]: Please put `-Wextra-semi` here to do not forget.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @undef21 , I left jsut one small comment here. Would you mind to check?

**Comment by ihhub** [comment]
@undef21 , thank you so much for cleaning up the code!

**Pull request merged**